docs/SCHEMA.md

```diff
@@ -0,0 +1,128 @@
+│       └──  snapshot
+│           └── ...
+├── debuggees
+│   ├── <debuggee id 1>
+│   │   └── <debuggee entity>
+│   ├── ....
+│   └── <debuggee id n>
+│       └── <debuggee entity>
+└── schema_version: "<version_number>"
+```
+
+## cdbg/breakpoints
+
+This node contains the breakpoints for debuggees stored on a per debuggee id
+basis. The breakpoints per debuggee are grouped into `active`, `final` and
+`snapshot`.
+
+To note, breakpoints can represent either a `Snapshot` or a `Logpoint`. The
+value of the `action` field of the breakpoint makes this determination, which
+will be either `CAPTURE` or `LOG`. If the `action` field is not populated the
+agents will default to `CAPTURE`.
+
+
+### cdbg/breakpoints/debuggee_id/active
+
+This node contains all of the active breakpoints for a given debuggee. When a
+breakpoint is created it is considered active. It remains active until it is
+either deleted by the user or it is finalized by an agent.
+
+Then agents will finalize a breakpoint under the following scenarios:
+
+1. A newly created breakpoint fails initial validation and is finalized being
+   marked with an error. One reason for this would be an inability to set a
+   breakpoint at the specified location, which may not exist.
+1. The breakpoint is a `Snapshot` and it triggers.
+1. The breakpoint is a `Snapshot` and it expires before being hit (by default
+   after 24 hours).
+1. The breakpoint is a `Logpoint` and it expires (by default after 24 hours).
+
+When the agent finalizes a breakpoint it will be removed from the `active` node
+and added to the `final` node, and possibly the `snapshot` node, see below for
+further details.
+
+### cdbg/breakpoints/debuggee_id/final
+
+This node contains all finalized breakpoints (both `Snapshot` and `Logpoint`)
+until deleted by the user. One important point to note, for `Snapshot`
+breakpoints, the `stackFrames`, `variableTable` and `evaluatedExpressions` are
+not present in the breakpoints under this node, it is the responsibility of the
+agent to ensure they do not populate these fields for this node. This is to make
+it cheaper to read all final breakpoints. When users need the full snapshot
+data, this can then be retrieved using the complete data found under the
+`snapshot` node.
+
+Also to note, it is the agent's responsibility to set the `finalTimeUnixMsec`
+field.
+
+### cdbg/breakpoints/debuggee_id/snaphsot
+
+This node contains all finalized `Snapshot` breakpoints until deleted by the
+user. For finalized `Snapshot` breakpoints that have capture data, the full data
+is present here, meaning the `stackFrames`, `variableTable` and
+`evaluatedExpressions` are present, unlike under the `final` node.
+
+### Breakpoint Entity
+
+The breakpoint entity is fully documented in protobuf form
+[here][breakpoint_entity]. The data in the Firebase DB is in JSON format, which
+protobuf maps to as described [here][json_mapping]. One thing to note
+is how Firebase stores [arrays][firebase_arrays].
```

Comment:
I'd move this up to the top, before we go into the active, final, snapshot details.  Those details refer to members of the proto, so it's good to have the proto up front.

Reply:
Done.